rpcserver.go

```diff
@@ -2172,28 +2172,24 @@ func (r *rpcServer) parseOpenChannelReq(in *lnrpc.OpenChannelRequest,
 
 	var feeRate chainfee.SatPerKWeight
 
-	// Skip estimating fee rate for PSBT funding.
-	if in.FundingShim == nil || in.FundingShim.GetPsbtShim() == nil {
-		// Keep the old behavior prior to 0.18.0 - when the user
-		// doesn't set fee rate or conf target, the default conf target
-		// of 6 is used.
-		targetConf := maybeUseDefaultConf(
-			in.SatPerByte, in.SatPerVbyte, uint32(in.TargetConf),
-		)
-
-		// Calculate an appropriate fee rate for this transaction.
-		feeRate, err = lnrpc.CalculateFeeRate(
-			uint64(in.SatPerByte), in.SatPerVbyte,
-			targetConf, r.server.cc.FeeEstimator,
-		)
-		if err != nil {
-			return nil, err
-		}
+	// NOTE: We also need to do the fee rate calculation for the psbt
+	// funding flow because the `batchfund` depends on it.
```

Comment:
I would leave it there because the current design is still not perfect and a bit more information is beneficial in my view.

Reply:
ah ok, I see that now, I wonder what would happen if the async channel open api as given different targets/or fee rates, so a uniform one should be enforced at some point

Reply:
I was wondering what would happen if https://lightning.engineering/api-docs/api/lnd/lightning/open-channel is called with different fee rates successively, trying to batch open, whether it would error out, but it's not related to the PR, so will resolve this